autoload/go/test.vim

```diff
@@ -253,28 +253,50 @@ endfunction
 
 function! s:parse_errors(lines) abort
   let errors = []
+  let paniced = 0 " signals whether all remaining lines should be included in errors.
 
   " NOTE(arslan): once we get JSON output everything will be easier :)
   " https://github.com/golang/go/issues/2981
   for line in a:lines
-    let fatalerrors = matchlist(line, '^\(fatal error:.*\)$')
-    let tokens = matchlist(line, '^\s*\(.\{-}\.go\):\(\d\+\):\s*\(.*\)')
-
+    let fatalerrors = matchlist(line, '^\(\(fatal error\|panic\):.*\)$')
     if !empty(fatalerrors)
-      call add(errors, {"text": fatalerrors[1]})
-    elseif !empty(tokens)
+      let paniced = 1
+      call add(errors, {"text": line})
+      continue
+    endif
+
+    let tokens = []
+    if paniced
+      let tokens = matchlist(line, '^\t\+\(.\{-}\.go\):\(\d\+\) \(+0x.*\)')
+    else
+      let tokens = matchlist(line, '^ *\t\+\(.\{-}\.go\):\(\d\+\):\s*\(.*\)')
+    endif
```

Comment:
Could you add as a comment above the `let tokens ...` lines what exactly it parses? Such as : `foo.go:1:15 msg` It's would make reading the regexp easier.

Reply:
Done